System/Posix/Terminal/Common.hsc

```diff
@@ -581,119 +768,61 @@ cc2Word Stop      = (#const VSTOP)
 -- Convert Haskell BaudRate to unsigned integral type (Word)
 
 baud2Word :: BaudRate -> CSpeed
-baud2Word B0 = (#const B0)
-baud2Word B50 = (#const B50)
-baud2Word B75 = (#const B75)
-baud2Word B110 = (#const B110)
-baud2Word B134 = (#const B134)
-baud2Word B150 = (#const B150)
-baud2Word B200 = (#const B200)
-baud2Word B300 = (#const B300)
-baud2Word B600 = (#const B600)
-baud2Word B1200 = (#const B1200)
-baud2Word B1800 = (#const B1800)
-baud2Word B2400 = (#const B2400)
-baud2Word B4800 = (#const B4800)
-baud2Word B9600 = (#const B9600)
-baud2Word B19200 = (#const B19200)
-baud2Word B38400 = (#const B38400)
-#ifdef B57600
-baud2Word B57600 = (#const B57600)
-#else
-baud2Word B57600 = error "B57600 not available on this system"
-#endif
-#ifdef B115200
-baud2Word B115200 = (#const B115200)
-#else
-baud2Word B115200 = error "B115200 not available on this system"
-#endif
-#ifdef B230400
-baud2Word B230400 = (#const B230400)
-#else
-baud2Word B230400 = error "B230400 not available on this system"
-#endif
-#ifdef B460800
-baud2Word B460800 = (#const B460800)
-#else
-baud2Word B460800 = error "B460800 not available on this system"
-#endif
-#ifdef B500000
-baud2Word B500000 = (#const B500000)
-#else
-baud2Word B500000 = error "B500000 not available on this system"
-#endif
-#ifdef B576000
-baud2Word B576000 = (#const B576000)
-#else
-baud2Word B576000 = error "B576000 not available on this system"
-#endif
-#ifdef B921600
-baud2Word B921600 = (#const B921600)
-#else
-baud2Word B921600 = error "B921600 not available on this system"
-#endif
-#ifdef B1000000
-baud2Word B1000000 = (#const B1000000)
-#else
-baud2Word B1000000 = error "B1000000 not available on this system"
-#endif
-#ifdef B1152000
-baud2Word B1152000 = (#const B1152000)
-#else
-baud2Word B1152000 = error "B1152000 not available on this system"
-#endif
-#ifdef B1500000
-baud2Word B1500000 = (#const B1500000)
-#else
-baud2Word B1500000 = error "B1500000 not available on this system"
-#endif
-#ifdef B2000000
-baud2Word B2000000 = (#const B2000000)
-#else
-baud2Word B2000000 = error "B2000000 not available on this system"
-#endif
-#ifdef B2500000
-baud2Word B2500000 = (#const B2500000)
-#else
-baud2Word B2500000 = error "B2500000 not available on this system"
-#endif
-#ifdef B3000000
-baud2Word B3000000 = (#const B3000000)
-#else
-baud2Word B3000000 = error "B3000000 not available on this system"
-#endif
-#ifdef B3500000
-baud2Word B3500000 = (#const B3500000)
-#else
-baud2Word B3500000 = error "B3500000 not available on this system"
-#endif
-#ifdef B4000000
-baud2Word B4000000 = (#const B4000000)
-#else
-baud2Word B4000000 = error "B4000000 not available on this system"
-#endif
+baud2Word = coerce
 
 -- And convert a word back to a baud rate
 -- We really need some cpp macros here.
 
 word2Baud :: CSpeed -> BaudRate
 word2Baud x = case x of
```

Comment:
This function should accept any number, whether known at the time the package was released, or not, and should not throw errors by default.  If the value is not actually supported by the terminal driver, the user will find out when the relevant ioctl fails.

Reply:
We should however also provide sensible `Show` and `Read` instances for these, examples can be found in the `network` package.

Reply:
I agree.

Reply:
With #175 merged, this PR can resume progress.  But we do need to sort out my prior comments. Are you ready/willing to get this back on track?  If not, I could perhaps take it over and get it across the line?